packages/components/src/sandbox/index.js

```diff
@@ -28,7 +28,11 @@ class Sandbox extends Component {
 		this.trySandbox();
 	}
 
-	componentDidUpdate() {
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.html !== this.props.html ) {
+			// Allows the new html to go into the sandbox.
+			this.iframe.current.contentDocument.body.removeAttribute( 'data-resizable-iframe-connected' );
```

Comment:
For folks reading there's an early exit in sandbox otherwise https://github.com/WordPress/gutenberg/blob/6f7fc0990cb87b8f71ecfac4f1ee18a040fea027/packages/components/src/sandbox/index.js#L75